gslib/utils/copy_helper.py

```diff
@@ -3682,14 +3682,14 @@ def _WriteRowToManifestFile(self, url):
     """Writes a manifest entry to the manifest file for the url argument."""
     row_item = self.items[url]
     data = [
-        row_item['source_uri'],
-        row_item['destination_uri'],
+        six.ensure_str(row_item['source_uri']),
+        six.ensure_str(row_item['destination_uri']),
         '%sZ' % row_item['start_time'].isoformat(),
         '%sZ' % row_item['end_time'].isoformat(),
         row_item['md5'] if 'md5' in row_item else '',
         row_item['upload_id'] if 'upload_id' in row_item else '',
-        str(row_item['size']) if 'size' in row_item else '',
-        str(row_item['bytes']) if 'bytes' in row_item else '',
+        six.ensure_str(str(row_item['size'])) if 'size' in row_item else '',
```

Comment:
(Here and potentially elsewhere)

Shouldn't `ensure_str` wrap around the entire rest of the expression?  If we have `unicode_literals` imported, the "else" case will give back a unicode object in PY2, but the "if" case will give back a `bytes` object.

Reply:
Ah, that makes sense!

Reply:
I ended up going back to list comprehension if that's ok :) newer comprehension tested good with Python 2.7 and 3.5 running cp tests on json / linux, doing full tests now

Reply:
Also, as we discussed offline, the `str()` cast inside of `ensure_str()` is sometimes necessary -- I think `ensure_str` expects the argument to be either `unicode` or `bytes`, so you'll need to cast things like "size" (an int) to a string-like thing first before passing it to `ensure_str()`.